### PR TITLE
[BZ1551752]: Update a secret

### PR DESCRIPTION
**Pull request by xenolinux**
OCP version: 3.11
[Preview](https://deploy-preview-41757--osdocs.netlify.app/openshift-enterprise/latest/dev_guide/secrets.html#secrets-and-the-pod-lifecycle)
[Bug]( https://bugzilla.redhat.com/show_bug.cgi?id=1551752)
QE contact: @wangke19

### CODE REVIEW

File: dev_guide/secrets.adoc [hunk 1]
```diff
@@ -160,7 +160,7 @@ For examples of differet secret types, see the xref:dev-guide-secrets-using-secr
 
 When you modify the value of a secret, the value (used by an already running
 pod) will not dynamically change. To change a secret, you must delete the
-original pod and create a new pod (perhaps with an identical PodSpec).
+original pod and create a new pod (perhaps with an identical PodSpec). When you mount a secret as a volume, your secret automatically gets updated.
```
[the remark this answers is not in the document]
Comment (wangke19) [reply dev_guide/secrets.adoc:163]: It is betterto add this topic to https://deploy-preview-41757--osdocs.netlify.app/openshift-enterprise/latest/dev_guide/secrets.html#secrets-in-volumes-and-env-vars,  there environment variables and secret are involved.

[the remark this answers is not in the document]
Comment (xenolinux) [reply dev_guide/secrets.adoc:163]: Ack! @xingxingxia @wangke19 can you please check and let me know if it looks good needs any change?

[the remark this answers is not in the document]
Comment (xenolinux) [reply dev_guide/secrets.adoc:163]: I added [this topic]( https://kubernetes.io/docs/concepts/configuration/secret/#environment-variables-are-not-updated-after-a-secret-update ) info [here](https://deploy-preview-41757--osdocs.netlify.app/openshift-enterprise/latest/dev_guide/secrets.html#secrets-in-volumes-and-env-vars).

File: dev_guide/secrets.adoc [hunk 2]
```diff
@@ -181,7 +181,7 @@ but create new ones with distinct names.
 [[secrets-in-volumes-and-env-vars]]
 == Secrets in Volumes and Environment Variables
 
-See xref:secrets-examples[examples] of YAML files with secret data.
+If your `secret` uses an environment container, you must restart the container to see the updated `secret`. See xref:secrets-examples[examples] of YAML files with secret data.
```
Comment (xingxingxia) [inline dev_guide/secrets.adoc:184]: Not secret uses env container. Rather:
Should be: If your container uses secret as environment.
Or copy the k8s doc: If a container already consumes a Secret in an environment variable

Comment (xenolinux) [reply dev_guide/secrets.adoc:184]: Going with the - ` Should be: If your container uses secret as environment.`

File: dev_guide/secrets.adoc [hunk 3]
```diff
@@ -181,7 +181,7 @@ but create new ones with distinct names.
 [[secrets-in-volumes-and-env-vars]]
 == Secrets in Volumes and Environment Variables
 
-See xref:secrets-examples[examples] of YAML files with secret data.
+If your container uses a `secret` in an environment container, you must restart the container to see the updated `secret`. See xref:secrets-examples[examples] of YAML files with secret data.
```
Comment (xingxingxia) [inline dev_guide/secrets.adoc:184]: Not sure what does `in an environment container` mean, seems not good wording. Maybe just:
```
If your container uses a `secret` as environment variable
```

Comment (xenolinux) [reply dev_guide/secrets.adoc]: Done

**Comment by xingxingxia** [comment]
/lgtm
Thanks

File: dev_guide/secrets.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (xenolinux) [reply dev_guide/secrets.adoc:163]: Acked

**Pull request merged**